daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/processors/unparsers/SuppressableSeparatorUnparser.scala

```diff
@@ -0,0 +1,268 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.daffodil.processors.unparsers
+
+import org.apache.daffodil.processors.SuspendableOperation
+import org.apache.daffodil.util.LogLevel
+import org.apache.daffodil.processors.TermRuntimeData
+import org.apache.daffodil.exceptions.Assert
+import org.apache.daffodil.util.Maybe
+import org.apache.daffodil.util.Maybe._
+import org.apache.daffodil.io.DataOutputStream
+import org.apache.daffodil.io.ZeroLengthStatus
+import scala.collection.mutable.Buffer
+import org.apache.daffodil.processors.Processor
+import org.apache.daffodil.io.DirectOrBufferedDataOutputStream
+
+final class SuppressableSeparatorUnparserSuspendableOperation(sepUnparser: Unparser,
```

Comment:
Add performance comment about special-purpose suspension. Unlike the suspension for an outputValueCalc, the expressions for seps can only refer backwards, so can be evaluated when encountered, the separator text squirreled away, and on unblocking, we decide to just blat out that text, or not. But no state other than said text to blat out is needed to be kept, so these can be much much lower overhead.